### DISCUSSION TITLE
Tif File with alpha => JPEG Compress ?!

### DISCUSSION THREAD

**Discussion by chukanet** [libvips/libvips#3252] [Help wanted]
Hi, i'm looking a simple operation but after 2 days i don't get a good result !

i got a TIF file with an alpha channel, and i just want to compress it with JPEG compress.

i work with the Command (CMD) only, and they're is some of my try:

vips tiffsave "C:\Users\Myname\Desktop\tif\test\test.tif" "C:\Users\Myname\Desktop\tif\test\x.tif" --compression jpeg --Q 60 --pyramid --tile

but the Alpha is flattened to the RGB... my file size looks ok but not the result...

i've tested : 
vips Unpremultiply "C:\Users\Myname\Desktop\tif\test\test.tif" "C:\Users\Myname\Desktop\tif\test\x.tif"[compression=jpeg,Q=60]

but i can see a little bit the alpha channel into the RGB...

i've tried this too : 

vips extract_band "C:\Users\Myname\Desktop\tif\test\test.tif" "C:\Users\Myname\Desktop\tif\test\rgb.tif"[compression=jpeg,tile] 0 --n 3
vips extract_band "C:\Users\Myname\Desktop\tif\test\test.tif" "C:\Users\Myname\Desktop\tif\test\alpha.tif"[compression=jpeg,tile] 3 --n 1

this extract the RGB and the Alpha into 2 separated file but how can i relink them into single file?
i've try :

vips bandjoin "C:\Users\Myname\Desktop\tif\test\rgb.tif" "C:\Users\Myname\Desktop\tif\test\alpha.tif" "C:\Users\Myname\Desktop\tif\test\Final.tif"[compression=jpeg,tile]

but i get some error...

Does someone could help me to get a perfect result ? and does it's possible to do this task in all Tif in a specific folder or i need to do it for each file?
Maybe it does exist a simple way to do ?! 

Thx a lot

**Comment by jcupitt** [comment]
Hi @chukanet,

I don't think libtiff supports this by default. JPEG compression is delegated to libjpeg, and it does not support alpha.

**Reply by jcupitt** [reply]
Oh, you could probably do it as a plane-separated image. libjpeg could then compress as four monochrome images. Although JPEG compression on alpha can produce poor results.

Unfortunately libvips does not support plane separated write -- you'll need to find some other software for this.

**Reply by chukanet** [reply]
Hi and thx for you help ^^ finally i've done it with imageMagick and it's work pretty well.
i've reduced my tif more than 12x and the quality looks ok... 
but some program doesn't accept JPEG compression

regards
